edge-functions/rewrite-to-external-url/pages/_middleware.js

```diff
@@ -0,0 +1,17 @@
+import { NextResponse } from 'next/server'
+
+export async function middleware(req) {
+  const { pathname } = req.nextUrl
+  const hostname = req.headers.get('host')
+
+  if (hostname === 'support-site.vercel.pub' || hostname === 'localhost:3000') {
+    // rewrites "support-site.vercel.pub" to "https://www.zendesk.com"
+    // note that it's important to check if the URL that you are rewriting to
+    // has trailing slashes or not – or else you'll end up with a redirect loop.
+    // also, note that the URL that you are rewriting to must be a full URL,
+    // and has to include the `www` subdomain if it's present.
+    return NextResponse.rewrite(`https://www.zendesk.com/${pathname}/`)
+  }
+  // perform the rewrite anyways (just for this demo)
+  return NextResponse.rewrite(`https://www.zendesk.com/${pathname}/`)
```

Comment:
I don't think this is needed, looks like an escape hatch if the above fails and it should never fail